apps/api/src/app/subscribers/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts

```diff
@@ -55,16 +57,15 @@ export class UpdateSubscriberPreference {
         channels: command.channel?.type ? channelObj : null,
       });
 
-      const template = await this.notificationTemplateRepository.findById(command.templateId, command.environmentId);
-
-      const getSubscriberPreferenceCommand = GetSubscriberTemplatePreferenceCommand.create({
-        organizationId: command.organizationId,
-        subscriberId: command.subscriberId,
-        environmentId: command.environmentId,
-        template,
-      });
-
-      return await this.getSubscriberTemplatePreference.execute(getSubscriberPreferenceCommand);
+      return await this.getSubscriberTemplatePreference.execute(
+        GetSubscriberTemplatePreferenceCommand.create({
+          organizationId: command.organizationId,
+          subscriberId: command.subscriberId,
+          environmentId: command.environmentId,
+          template,
+          skipCache: true,
```

Comment:
This command needs to skip findOne because the create method above won't invalidate in time.

Reply:
I'm not sure if this is a good solution to the problem you hit, because it moves the `knowledge` about the caching from the DAL to the Use Cases level.

The real problem hides in the `CacheService.delByPattern` function, as it doesn't wait for the `key` to be deleted before moving to the next statement.
Like the `create` function is called which tries to invalidate the cache with `delByPattern` but it doesn't wait for the result which is `pipeline.exec` promise to finish executing.

I would recommend to try to wrap the `delByPattern` with a promise like:
```
public async delByPattern(pattern: string) {
    return new Promise((resolve, reject) => {
      const client = this.client;
      const stream = client.scanStream({
        match: pattern,
      });

      stream.on('data', function (keys) {
        if (keys.length) {
          const pipeline = client.pipeline();
          keys.forEach(function (key) {
            pipeline.del(key);
          });
          pipeline.exec().then(resolve).catch(reject);
        }
      });
    });
  }
```

Then you will need to `await` for it in the `InvalidateCache`
```
await this.cacheService.delByPattern(cacheKey);
```

Please let me know if it works.